src/ring_theory/algebra_tower.lean

```diff
@@ -194,6 +184,53 @@ le_antisymm (adjoin_le $ set.image_subset_iff.2 $ λ y hy, ⟨y, subset_adjoin h
 
 end algebra
 
+namespace subalgebra
+
+open is_algebra_tower
+
+variables (R) {S A} [comm_semiring R] [comm_semiring S] [semiring A]
+variables [algebra R S] [algebra S A] [algebra R A] [is_algebra_tower R S A]
+
+/-- If A/S/R is a tower of algebras then any S-subalgebra of A gives an R-subalgebra of A. -/
+def res (U : subalgebra S A) : subalgebra R A :=
+{ carrier := U,
+  algebra_map_mem' := λ x, by { rw algebra_map_apply R S A, exact U.algebra_map_mem _ } }
+
+@[simp] lemma res_top : res R (⊤ : subalgebra S A) = ⊤ :=
+algebra.eq_top_iff.2 $ λ _, show _ ∈ (⊤ : subalgebra S A), from algebra.mem_top
```

Comment:
What is the reason for renaming these? (Actually, what does `res` stand for?)

Reply:
shorter; restrict.